tcf_website/static/department/sortClasses.js

```diff
@@ -1,67 +1,104 @@
+        $("#gpa-sort-btn").html("GPA");
         $("#number-sort-btn").addClass("active");
+        // Choose default order
     } else { // Already selected so simply reverse the order of sorting (ascending vs descending)
         sortNumberAsc *= -1;
     }
-
-    // Actual sorting
-    sortClasses(cmpByProp("title", sortNumberAsc));
+    selectOrderOfSort(sortNumberAsc, "#number-sort-btn", "Course ID");
 };
 
 const sortByRating = () => {
+    activeSort = "rating";
     if (!$("#rating-sort-btn").hasClass("active")) {
         $("#number-sort-btn").removeClass("active");
         $("#diff-sort-btn").removeClass("active");
         $("#gpa-sort-btn").removeClass("active");
+        $("#number-sort-btn").html("Course ID");
+        $("#diff-sort-btn").html("Difficulty");
+        $("#gpa-sort-btn").html("GPA");
         $("#rating-sort-btn").addClass("active");
     } else {
         sortRatingAsc *= -1;
     }
-
-    sortClasses(cmpByProp("rating", sortRatingAsc));
+    selectOrderOfSort(sortRatingAsc, "#rating-sort-btn", "Rating");
 };
 
 const sortByDifficulty = () => {
+    activeSort = "difficulty";
     if (!$("#diff-sort-btn").hasClass("active")) {
         $("#number-sort-btn").removeClass("active");
         $("#rating-sort-btn").removeClass("active");
+        $("#gpa-sort-btn").html("GPA");
+        $("#number-sort-btn").html("Course ID");
+        $("#rating-sort-btn").html("Rating");
         $("#gpa-sort-btn").removeClass("active");
+
         $("#diff-sort-btn").addClass("active");
     } else {
         sortDifficultyAsc *= -1;
     }
-
-    sortClasses(cmpByProp("difficulty", sortDifficultyAsc));
+    selectOrderOfSort(sortDifficultyAsc, "#diff-sort-btn", "Difficulty");
 };
 
 const sortByGpa = () => {
+    activeSort = "gpa";
     if (!$("#gpa-sort-btn").hasClass("active")) {
         $("#number-sort-btn").removeClass("active");
         $("#rating-sort-btn").removeClass("active");
         $("#diff-sort-btn").removeClass("active");
+        $("#number-sort-btn").html("Course ID");
+        $("#rating-sort-btn").html("Rating");
+        $("#diff-sort-btn").html("Difficulty");
         $("#gpa-sort-btn").addClass("active");
     } else {
         sortGpaAsc *= -1;
     }
+    selectOrderOfSort(sortGpaAsc, "#gpa-sort-btn", "GPA");
+};
 
-    sortClasses(cmpByProp("gpa", sortGpaAsc));
+/* Helper function used by all sort functions to:
+   -Compute the relevant sort based on what sort button is active
+   -Select the relevant ascending/descending arrow and add it as text
+*/
+const selectOrderOfSort = (asc, id, sortName) => {
+    /* sortClasses takes in a comparing function (defined below) and reorders the classes appropriately
+       comparator needed for both ascending and descending as any classes with null values should be at the end of the list either order
+    */
+    var arrow = "";
+    sortClasses(cmpByProp(activeSort, asc));
+    if (asc === 1) {
+        arrow = " ⬆";
+    } else {
+        arrow = " ⬇";
```

Comment:
Minor nitpick, but not sure if it's better to use the Unicode hex value instead For example, 
```
arrow = " \u0394" # Up arrow (dummy Unicode value replace with the real one)
```
It's more of a personal preference thing so up to you if you want to change. If you do though, make sure to do the same in the template code for consistency

Reply:
I agree as well. It'll get rid of any inconsistencies in the future if the arrow doesn't show up on certain computers and os systems etc.